### DISCUSSION TITLE
Montage with overlap. Negative offset crops edges

### DISCUSSION THREAD

**Discussion by oliverpeng** [ImageMagick/ImageMagick#2330] [Help] [answered]
I'm trying to vertically tile images that have 10 pixels of overlap. However, when I try the following command, the top and bottom edges also get clipped, but I would like to preserve them:

montage red.jpg blue.jpg green.jpg -geometry +0-5 -tile 1x output.png

Any help greatly appreciated!

ImageMagick 7.0.10-24
OS X

![montage](https://user-images.githubusercontent.com/1031384/88560144-c231b580-cffb-11ea-92b5-85746d823837.jpg)

**Comment by snibgo** [comment]
According to your picture, you want each overlap to contain some mixture of colours from the two inputs. Is that what you want?

Please show your input files, or the IM commands that make them.

**Reply by oliverpeng** [reply]
No, the overlaps are identical between the squares (think map tiles). The mixed colors is just to visually show that they overlap; I don't need to blend anything. The overlapping data can come from any of the squares as it is duplicated data. Unfortunately I cannot post the input files as they are not owned by me but I hope this clarifies the situation somewhat. Thank you for your assistance!

**Comment by snibgo** [comment]
I'm not sure what you want. It may be easier to use "magick", not "montage". For example, Windows BAT syntax:

```
magick ^
  r.png ^
  ( b.png -repage +0+180 ) ^
  ( g.png -repage +0+360 ) ^
  -layers merge ^
  out.png
```
This places the r.png image. Over that, with 20 pixel overlap, it places g.png, so the bottom 20 pixels of r.png are hidden. Then it puts g.png, hiding the bottom part of g.png.

**Comment by GeeMack** [comment, marked as answer]
If you're only appending the pieces vertically, you may be able to get away with "magick" and "-smush" instead of "montage" to get a 10 pixel overlap. This command creates a red, a blue, and a green square, then "-smush"es them with an argument of "-10" to make the vertically stacked output image... 

     magick -size 400x400 xc:red xc:blue xc:green -smush -10 result.png

Use "-smush" with positive numbers to leave a gap between the images. Using the plus "+" form of "+smush" will append the pieces horizontally.

**Reply by oliverpeng** [reply]
Thank you, this did the trick!

**Reply by oliverpeng** [reply]
After building two vertical columns, is there a way I can smush two of those together without having to create intermediary files? End result being a 2x3 grid. The horizontal smushing doesn't require any offsets, just simply glueing the edges together.

**Reply by GeeMack** [reply]
With ImageMagick you can work separate operations isolated inside parentheses Look at this command to see one way to make that 2x3 grid...

    magick \( i1.png i2.png i3.png -smush -10 \) \ 
        \( i4.png i5.png i6.png -smush -10 \) +append result.png

With ImageMagick there are usually several approaches to any task. This command uses "-crop" and "-append" and "-smush" to get that same result.

    magick img1.png img2.png img3.png img4.png img5.png img6.png \
        -append -crop 1x2@ +append -crop 1x3@ -smush -10 result.png

**Reply by oliverpeng** [reply]
Thank you, this was immensely helpful. Deeply appreciated!
